# Romteknologi II - Hjemmeoppgaver April 2018

## Skrevet av Øyvind Aukner

## Oppgave 1

1. Solen gjennomgår sykler på cirka 11 år. Gjennom grundig observasjon kan man forutsi når solen vil være på sitt mest, og minst aktive. Dette har konsekvenser for hvor mye stråling, og da også forstyrrelse som vil ramme Jorda. Generelt "romvær" er nyttig viten når man skal forvalte presise instrumenter plassert I baner og overgangsbaner.
2. Gjennom tidlig historie har stjerne- og spesielt solobservasjon latt sapiens forutsi årstider og andre klimatiske trender. Det er minst like sant den dag i dag, bare på mer detaljnivå.
3. Sola er en ganske gjennomsnittlig stjerne. Ved å observere den kan vi lære mye om stjerner generelt i rommet.

## Oppgave 2

### L1 Lagrange punkt
L1 er ett av totalt fem punkt hvor man finner balanse mellom to store gravitasjonsbrønner. Som oftest snakker man om Jorda og Sola. I SOHO's tilfelle er L1 det punktet innenfor Jorda, inn mot Sola. For å intuetivt forstå punktet, kan man se for seg en balansegang. Nærme Jorda, faller man mot Jorda, og nærme Sola, faller man mot Sola. Ett eller annet sted I mellom, er det balanse og man faller i teorien ingensteder. Dette kalles L1 og er 1.5x10^6Km fra jordsentrum. Lagrangepunktene er i praksis ikke helt stabile ettersom månen og andre påvirker banen. Derfor må man i med noen få måneders mellomrom gjennopprette banen ved bruk av noe drivstoff. 

SOHO ble plassert her for å gi et uendrende perspektiv i Solas retning, som er det samme som oss her på jorda.

### SOHO's oppdrag
SOHO's oppdrag var å studere Sola, men har siden også oppdaget over 3000 kometer. Den hadde tre hovedoppdag:
1. se på Solas yttre lag, herunder coronaen samt opphavet til solvind
2. Studere egenskaper til L1 
3. Prøve å finne ut av Solas struktur. 

## Oppgave 3

Kommunikasjon med SOHO datt ut, og satellitten gikk inn i Emergency Sun Reacquisition (ESR) mode. Den endte opp med en liten spinn og det tok mange dager før man var i stand til å lokalisere og gjennnoppta forbindelse med satellitten. Gjennom ren og skjær flaks tillot rotasjonen noe sollys å treffe panelene i hver periode noe som tillot et vågalt og suksserikt redningsforsøk i de påfølgende ukene. Alle 12 instrumenter for datasamling ble brukbare igjen, mens kontrollgyroene aldri kom tilbake. Dette førte til en tidsperiode med dyre rakettdytteaktiveringer (thruster burns) før ekspertgruppa var i stand til å implimentere gyroløs kontroll 1.februar 1999 ved hjelp av programvare. Denne teknikken var banebrytene og har senere inspirert andre romfartøy.

## Oppgave 4

Vi starter med å legge inn et par bibliotek som lar oss regne med logaritmer og trigonometri, samt bruke benevning i fysiske størrelser

In [1]:
import math, physics

In [2]:
from physics import Q

In [3]:
%load_ext physics

Unit calculation and physics extensions activated.


In [4]:
π, c = physics._constants[0][1], physics._constants[2][1] # PI, Lysets hastighet i vakum

Først må vi legge inn verdiene og konvertere til SI

In [5]:
f, D = (2.25 GHz).base, Q(1.5*10**6, 'km').base # .base operatoren konverterer til SI hhv. Hz og m

In [6]:
λ = c/f

In [7]:
f, D, c, λ

(2.25e+09 1/s, 1.5e+09 m, 2.9979246e+08 m/s, 0.13324109 m/1)

Formelen for å regne ut **Free Space Loss** er: 
> <font size="5px">$L_{0} = (\frac{4πD}{λ})^{2}$</font>


In [8]:
L0 = (4*π*D/λ)**2

In [9]:
L0

2.0013630459705637e+22

For å få **verdien i desibel** må vi ta: 
> <font size="5px">$10 \times Log_{10}(Lineærverdi)$:</font>

In [10]:
L0_db = 10*math.log10(L0)

In [11]:
L0_db

223.01325876522426

> ### **Free Space Loss for SOHO er 223 dB**

## Oppgave 5

Systemtemperatur har mye å si for radiostøy. Antennetemperatur er en parameter som beskriver hvor mye støy en antenne produserer i et gitt miljø. Dette kalles ofte antennesøty temperatur (antenna noise temperature) og er ikke et mål på antennas fysiske temperatur. Derimot avhenger den av dens _"gain pattern"_ og det termiske miljøet den er utplassert i. 

Opphavet er at temperaturen er en temperaturen er temperaturen på en hypotesisk resistor med input av ideel _"noise free receiver"_ som vil generere samme _"output noise power per unit bandwidth"_ som antennen ved en spesifikk frekvens.

For å bedre signalet ønsker vi å redusere systemtemperaturen. Effekten denne har er altså mer negativ ved høyere verdier.

## Oppgave 6

Kort sagt, fordi satelliten ikke er akuratt i L1 og Sola er ekstremt mye lenger unna satelitten enn det Jorda er. 

Det lange svaret er at Sola er en enorm kilde til termisk stråling og radiostøy. Det vil si at strålespekteret utstrålt av Sola også dekker båndet brukt av satellitten. En eksakt L1 posisjon ville altså vært vannskelig å kommunisere med fordi for å peke en mottakerantenne på jorda mot SOHO, måtte vi også pekt rett på Sola. Men om vi lar satellitten ta en liten omvei rundt L1 unngår vi dette problemet. SOHO er derfor ikke akuratt på L1, men i en såkalt Lissajous bane rundt den som vist i figuren under. Denne banen har en periode på seks måneder, mens rundt Sola går det naturligvis et år per periode. 

![alt text][SOHO mission schematic]

[SOHO mission schematic]: http://images.slideplayer.com/25/7826609/slides/slide_6.jpg "Logo Title Text 2"

## Oppgave 7

## Transmitter

Først noen ingangsverdier:

In [12]:
Pt, antenna_diameter, antenna_efficiency = Q(3.2, 'W'),  Q(0.8, 'm'), 0.8 # Pt er Transmitter power

In [13]:
A = π*(antenna_diameter/2)**2 # Aperture

In [14]:
Pt_db = 10*math.log10(Pt.value) # Omregnet til desibel

Formelen for å omregne **antenna gain** er: 
> <font size="5px">$Gt = \frac{4πA}{λ^{2}}$</font>



In [15]:
Gt = antenna_efficiency * 4*π*A/λ**2

In [16]:
Gt_db = 10*math.log10(Gt)

In [17]:
print('Antenna gain i lineært: ' + str(Gt) + '\nAntenna gain i desibel: ' + str(Gt_db) + ' db')

Antenna gain i lineært: 284.6382998713691
Antenna gain i desibel: 24.542933367309683 db


Formelen for å regne ut **EiRP** er:
> <font size="5px">$Pt \times Gt$</font>

In [18]:
EiRP = Pt * Gt

In [19]:
EiRP_db = 10*math.log10(EiRP.value)

In [20]:
print('EiRP i lineært: ' + str(EiRP) + '\nEiRP i desibel: ' + str(EiRP_db) + ' db')

EiRP i lineært: 910.84256 W
EiRP i desibel: 29.594433150508745 db


## Receiver

In [21]:
antenna_diameter, antenna_efficiency = Q(26, 'm'),  0.9

In [22]:
A = π*(antenna_diameter/2)**2 # Aperture

Formelen for å regne ut **antenna gain** er:
> <font size="5px">$G_{r} = n \times \frac{4πA}{λ^{2}}$</font>

Hvor n er antenna efficiency og A er aperture

In [23]:
Gr = antenna_efficiency * 4*π*A/λ**2

In [24]:
Gr_db = 10*math.log10(Gr)

In [25]:
print('Antenna gain i lineært: ' + str(Gr) + '\nAntenna gain i desibel: ' + str(Gr_db) + ' db')

Antenna gain i lineært: 338230.3547690253
Antenna gain i desibel: 55.29212581136099 db


Formelen for **Recieved Power** er: 
> <font size="5px">$P_{r} = \frac{EiRP \times G_{r} }{ L_{0}}$</font>
    
Alle verdier er i desibel.

In [26]:
Pr_db = EiRP_db * Gr_db / L0_db

In [27]:
print('Recieved Power er ' + str(Pr_db) + ' dB')

Recieved Power er 7.337407336827839 dB


For å gå videre å finne noise power density og carrier to noise ratio trenger vi først å vite båndbredden. I slide 63 fra Satellittkommunikasjonsforelesningen står det at:

 - S/N = EsB/N0B
 - S/N er det samme som C/N
 - Da blir C/N0 = Es*B/N0 = Eb*mB/N0

Vi kan da sette opp følgende uttrykk:

> <font size="5px">$\frac{E_{b}mB}{N_{0}} = EiRP - L_{0} + \frac{Gr}{T} - k - B - L_{a}$</font>

m er et mål for koding, og siden vi nå skal se bort i fra BPKS så antar vi denne til 1. Den faller med andre ord bort. Hvis vi videre løser for B får vi:

> <font size="5px">$B = \frac{EiRP - L_{0} + \frac{Gr}{T} - k - L_{a}}{E_{b}/N_{0} + 1}$</font>

Alle disse verdiene er kjent. Så nå gjenstår kun å legge verdier og finne B:

In [28]:
k, T = physics._constants[13][1], Q(78, 'K') # boltzman's constant, Systemtemperatur

In [29]:
k_db, T_db, Eb_N0_db = 10*math.log10(k.value), 10*math.log10(T.value), 15

In [30]:
La_db = 10 # Additional losses

In [31]:
B_db = (EiRP_db - L0_db + (Gr_db/T_db) - k_db - La_db) / (Eb_N0_db + 1)

In [32]:
B = Q(10**(B_db/10), 'Hz') # Omregner til lineært

In [33]:
B, B_db

(1.4984468 Hz, 1.7564133042231056)

Formelen for å regne ut **Noise power density** er:
> <font size="5px">$N = kTB$</font>

Hvor k er boltzman's konstant, T er systemtemperatur og B er båndbredden

In [34]:
k, k_db, T, T_db, B

(1.3806488e-23 J/K,
 -228.59916780233414,
 78 K,
 18.920946026904804,
 1.4984468 Hz)

In [35]:
N = k * T * B # Noise power density

In [36]:
N_db = 10*math.log10(N.value)

In [37]:
N_db

-207.92180847120622

**Carrier to noise power density ratio C/N0** er det samme som signal to noise ratio og regnes ut for desibel på følgende måte S/N_db = EiRP-L0 + G/T - k - B - La. Alle verdier er i desibel

In [38]:
SNR_db = EiRP_db - L0_db + (Gr_db/T_db) - k_db - B_db - La_db # SNR er signal to noise ratio

In [39]:
SNR_db

26.346199563346588

Når m = 1 så vil det si 1 bit / symbol, altså er båndbredde og dataoverføringshastighet det samme. Vi får da en data_rate lik B:


In [40]:
data_rate = B

In [41]:
print(str(data_rate.value * 10**-3) + ' kbit/s')

0.0014984468062463137 kbit/s


Vi får altså en overføringshastighet på 0.001 kbit/s. Dette er utrolig lavt, og stemmer sannsynligvis ikke. Jeg mistenker at det er blitt gjort noe feil i utregning av båndbredden. Ofte lett å surre med lineære og desibel verdier. 

Oppgaven spør spesifikt om å besvare oppgaven ved å fylle ut excel arket, så under følger det:

![soho_excel](img/space_tech/soho_excel.png)

# Oppgave 8

SDO står for Solar Dynamics Observatory. Målet til SDO satellitten er å få en vitenskapelig forståelse for hvordan forbindelsen mellom Jorda og Sola påvirker liv og samfund her på Jorda. Man ønsker også å se på hvordan rommet nær Jorda blir påvirket av Solas nærvær. Videre studerer SDO Solas magnetfelt og hvordan det påvirker solvind og andre romvind-fenomener. Dette oppnåes blandt annet ved å studere mange bølgelengder samtidig. Forskere prøver å bruke satellitten til å forstå hvordan Solas magnetfelt oppstår og videre hvordan denne oppsamlede magnetenergien slippes it i rommet i form av solvind og energi. 

# Oppgave 9

Banen til SDO er _"inclined"_ geostasjonær og satt til 102* Vest. Denne banen er valgt for å gi kontinuerlig dekning av Sola samt relativt høy overføringshastighet til kun én bakkestasjon. Et alternativ var LEO, men dette ville ført til et behov for stor datalagringskapasitet. Man gikk heller for en kontinuerlig forbindelse. Når man tar det valget, vil det koste noe mer å skyte opp, da banen er lenger ut. Det vil også være noen perioder med jordskygge og måneskygge vært år. 

# Oppgave 10

Likt som sist legger vi inn verdiene og konvertere til SI:

In [42]:
f, D = (26.5 GHz).base, (35794 km).base

In [43]:
λ = c/f

In [44]:
L0 = (4*π*D/λ)**2

In [45]:
L0_db = 10*math.log10(L0)

In [46]:
f, D, c, λ

(2.65e+10 1/s, 35794000 m, 2.9979246e+08 m/s, 0.011312923 m/1)

In [47]:
L0, L0_db

(1.580849541401098e+21, 211.98890537580084)

> ### **Free Space Loss for SDO er 212 dB**

# Oppgave 11

## Transmitter

Først noen ingangsverdier:

In [48]:
Pt, antenna_diameter, antenna_efficiency = Q(2, 'W'),  Q(0.6, 'm'), 0.7 # Pt er Transmitter power

In [49]:
A = π*(antenna_diameter/2)**2 # Aperture

In [50]:
Pt_db = 10*math.log10(Pt.value) # Omregnet til desibel

In [51]:
Pt_db

3.010299956639812

Formelen for å omregne **antenna gain** er: 
> <font size="5px">$Gt = \frac{4πA}{λ^{2}}$</font>



In [52]:
Gt = antenna_efficiency * 4*π*A/λ**2

In [53]:
Gt_db = 10*math.log10(Gt)

In [54]:
print('Antenna gain i lineært: ' + str(Gt) + '\nAntenna gain i desibel: ' + str(Gt_db) + ' db')

Antenna gain i lineært: 19433.482258231696
Antenna gain i desibel: 42.88550628187572 db


Formelen for å regne ut **EiRP** er:
> <font size="5px">$Pt \times Gt$</font>

In [55]:
EiRP = Pt * Gt

In [56]:
EiRP_db = 10*math.log10(EiRP.value)

In [57]:
print('EiRP i lineært: ' + str(EiRP) + '\nEiRP i desibel: ' + str(EiRP_db) + ' db')

EiRP i lineært: 38866.965 W
EiRP i desibel: 45.89580623851553 db


## Receiver

In [58]:
antenna_diameter, antenna_efficiency = Q(18, 'm'),  0.8

In [59]:
A = π*(antenna_diameter/2)**2 # Aperture

Formelen for å regne ut **antenna gain** er:
> <font size="5px">$G_{r} = n \times \frac{4πA}{λ^{2}}$</font>

Hvor n er antenna efficiency og A er aperture

In [60]:
Gr = antenna_efficiency * 4*π*A/λ**2

In [61]:
Gr_db = 10*math.log10(Gr)

In [62]:
print('Antenna gain i lineært: ' + str(Gr) + '\nAntenna gain i desibel: ' + str(Gr_db) + ' db')

Antenna gain i lineært: 19988724.608466886
Antenna gain i desibel: 73.00785084604584 db


Formelen for **Recieved Power** er: 
> <font size="5px">$P_{r} = \frac{EiRP \times G_{r} }{ L_{0}}$</font>
    
Alle verdier er i desibel.

In [63]:
Pr_db = EiRP_db * Gr_db / L0_db

In [64]:
print('Recieved Power er ' + str(Pr_db) + ' dB')

Recieved Power er 15.806271419631853 dB


For å gå videre å finne noise power density og carrier to noise ratio trenger vi først å vite båndbredden. I slide 63 fra Satellittkommunikasjonsforelesningen står det at:

 - S/N = EsB/N0B
 - S/N er det samme som C/N
 - Da blir C/N0 = Es*B/N0 = Eb*mB/N0

Vi kan da sette opp følgende uttrykk:

> <font size="5px">$\frac{E_{b}mB}{N_{0}} = EiRP - L_{0} + \frac{Gr}{T} - k - B - L_{a}$</font>

m er et mål for koding, og siden vi nå skal se bort i fra BPKS så antar vi denne til 1. Den faller med andre ord bort. Hvis vi videre løser for B får vi:

> <font size="5px">$B = \frac{EiRP - L_{0} + \frac{Gr}{T} - k - L_{a}}{E_{b}/N_{0} + 1}$</font>

Alle disse verdiene er kjent. Så nå gjenstår kun å legge verdier og finne B:

In [65]:
k, T = physics._constants[13][1], Q(235, 'K') # boltzman's constant, Systemtemperatur

In [66]:
k_db, T_db, Eb_N0_db = 10*math.log10(k.value), 10*math.log10(T.value), 15

In [67]:
La_db = 15 # Additional losses

In [68]:
B_db = (EiRP_db - L0_db + (Gr_db/T_db) - k_db - La_db) / (Eb_N0_db + 1)

In [69]:
B = Q(10**(B_db/10), 'Hz') # Omregner til lineært

In [70]:
B, B_db

(2.0708917 Hz, 3.1615738331362007)

Formelen for å regne ut **Noise power density** er:
> <font size="5px">$N = kTB$</font>

Hvor k er boltzman's konstant, T er systemtemperatur og B er båndbredden

In [71]:
k, k_db, T, T_db

(1.3806488e-23 J/K, -228.59916780233414, 235 K, 23.710678622717364)

In [72]:
N = k * T * B # Noise power density

In [73]:
N_db = 10*math.log10(N.value)

In [74]:
N_db

-201.72691534648055

**Carrier to noise power density ratio C/N0** er det samme som signal to noise ratio og regnes ut for desibel på følgende måte S/N_db = EiRP-L0 + G/T - k - B - La. Alle verdier er i desibel

In [75]:
SNR_db = EiRP_db - L0_db + (Gr_db/T_db) - k_db - B_db - La_db # SNR er signal to noise ratio

In [76]:
SNR_db

47.42360749704301

Når m = 1 så vil det si 1 bit / symbol, altså er båndbredde og dataoverføringshastighet det samme. Vi får da en data_rate lik B:


In [77]:
data_rate = B

In [78]:
print(str(data_rate.value * 10**-3) + ' kbit/s')

0.0020708916800983943 kbit/s


Vi får altså en overføringshastighet på 0.002 kbit/s. Dette er dobbelt så mye som SOHO, noe som er logisk siden den er mye nærmere. Når det er sagt hadde jeg ventet enda mye mer, også i forhold til SOHO. Men er som sagt litt usikker på utregninga i link budgettet.

In [79]:
1/T

0.0042553191 1/K

In [80]:
10 * math.log10(B.value)

3.1615738331362007

![sdo_excel](img/sdo_excel.png)


# Polar Light
***

# Oppgave 12

## Trianguleringsmetoden

Trianguleringsmetoden som brukt til nordlysforskning går ut på å observere og ta bilder fra forskjellige lokasjoner og deretter regne ut høyde basert på observasjonsvinkelen. På figuren under kan man se inngangsvariablene for en slik utregning der *h* er den ukjente som man ønsker å finne.
![soho_excel](img/tri.png)

## Utregning

For å finne h må vi først finne x i trekanten ABC ved hjelp av sinussetningen. I vårt tilfelle er vinklene som målingene er tatt med på 75 og 78 grader. Og de er tatt med 5 km mellomrom. Sinussetningen kan lett regnes om til 
x = d * (sin∠B / sin∠C). Vinklene gir inngangsverdier til å regne ut alle vinkler da vi vet at de til sammen er 180°:

![soho_excel](img/tri_math.png)

In [81]:
d, sinB, sinC = Q(5, 'km'), math.sin(math.radians(102)), math.sin(math.radians(3))

In [82]:
x = d * (sinB / sinC)

In [83]:
d, sinB, sinC, x

(5 km, 0.9781476007338057, 0.052335956242943835, 93.448909 km)

For så å regne ut høyden *h* må vi bruke pytagoras i kjent stil for trekanten ADC: 
> <font size="5px">$sinA = \frac{mot}{hyp} \to mot = sinA \times hyp \to h = sinA \times x$</font>

In [84]:
sinA = math.sin(math.radians(75))

In [85]:
h = sinA * x

In [86]:
h

90.264714 km

> ### **Nordlysets høyde er i følge Størmes trianguleringsmetode 90.3 km over jordoverflaten**
> Det vil også si at nordlyset er relativt lavt, da nordlys vanligvis befinner seg mellom 100 km og 150 km over bakken. Gitt at Jorda er flat, men denne antakelsen er rimelig når avstanden kun er 5 km.
***

# Oppgave 13

Såkalt protonlys skjer som en følge av at energiprotoner, $H^{+}$, penetrerer atmosfæren og påvirker lyset. 

$M + H^{+} → M^{+} + H^{*}$

$H^{*} → H + hf$

$M + H → H^{+} + M + e_{n}$

Så lenge den gjenværende energien i protonet er over 36 eV, kan prosessen gjentaes.


# Oppgave 14

Ved 486.1 nm er lyset blågrønt og den elektroniske energien går fra n=4 til n=2. Den skifter altså to nivåer 4→2 og har en energiforskjell på 2.55 eV.

# Oppgave 15

Farten er overført til fotoner. Dopler shift på Δf/f=0.01. Hvor Δf er frekvensforskjellen pga dopler shift og f er frekvensen for det utstrålte fotonet

Formelen for doppler shift er i følge https://www4.uwsp.edu/physastr/kmenning/Astr311/Lect21.pdf :
> <font size="5">$\frac{Δλ}{λ} = \frac{v}{c}$</font>

In [87]:
doppler_shift = 0.01

Vi vet at bølgelengden er λ = 486.1 nm. Dette gir en rest frekvens:

In [88]:
fr = Q(486.1, 'nm').base # Rest frequency

Videre må vi regne om til bølgelengde:

In [89]:
λ = c / fr

In [90]:
num1 = doppler_shift * fr

In [91]:
λ, fr, num1

(6.1673001e+14 1/s, 4.861e-07 m, 4.861e-09 m)

In [92]:
v = c * (num1 / fr)

In [93]:
v

2997924.6 m/s

> ### Vi får en fart på nesten 3000 Km/s

Den relative farten regner vi enkelt ut

In [94]:
relative_speed = v / c

In [95]:
relative_speed

0.01

> ### Farten er ganske nøyaktig 1% av lysets hastighet

# Oppgave 16

Formelen for **kinetisk energi** er:
> <font size="5px">$E_{k} = \frac{mv^{2}}{2}$</font>

In [96]:
mass_proton, e0 = physics._constants[10][1], physics._constants[8][1]

In [97]:
Ek = (mass_proton * (v**2)) / 2

In [98]:
Ek.convert('J')

In [99]:
Ek

7.5163874e-15 J

Den kinetiske energien er 7.5e-15 J

In [100]:
Ek_eV = Ek * e0

In [101]:
Ek_eV

1.204258e-33 J*C

Den kinetiske energien er eV 1.2e-33 noe som ikke er skadelig for mennesker.

physics plugin docs: http://www.southampton.ac.uk/~fangohr/blog/physical-quantities-numerical-value-with-units-in-python.html
plugin for python 3: https://github.com/TheGrum/python3-physics
LateX math: https://en.wikibooks.org/wiki/LaTeX/Mathematics

In [102]:
physics._constants

[('pi', 3.141592653589793),
 ('e', 2.718281828459045),
 ('c0', 2.9979246e+08 m/s),
 ('mu0', 1.2566371e-06 kg*m/s^2/A^2),
 ('eps0', 8.8541878e-12 s^4*A^2/kg/m^3),
 ('Grav', 6.67384e-11 m^3/s^2/kg),
 ('hpl', 6.6260696e-34 s*J),
 ('hbar', 1.0545717e-34 s*J),
 ('e0', 1.6021766e-19 C),
 ('me', 9.1093829e-31 kg),
 ('mp', 1.6726218e-27 kg),
 ('mn', 1.6749274e-27 kg),
 ('NA', 6.0221413e+23 1/mol),
 ('kb', 1.3806488e-23 J/K),
 ('g0', 9.80665 m/s^2),
 ('R', 8.3144621 J/K/mol),
 ('alpha', 0.0072973525698),
 ('Ry', 10973732 1/m),
 ('mu_n', -9.6623647e-27 J/T),
 ('gamma', 183.24718 MHz/T),
 ('h0', 0.704),
 ('sigmaT', 6.652453e-29 m^2)]